File: coreutils/src/main/java/com/tanveershafeeprottoy/coreutils/TransitionUtils.java
package com.tanveershafeeprottoy.coreutils;

import android.transition.ChangeBounds;
import android.transition.Explode;
import android.transition.Fade;
import android.transition.Transition;
import android.view.animation.DecelerateInterpolator;

/**
 * @author Tanveer Shafee Prottoy
 */

public class TransitionUtils {
    private static ChangeBounds changeBounds;

    public static Transition changeBounds(long duration) {
        changeBounds = new ChangeBounds();
        changeBounds.setDuration(duration);
        return changeBounds;
    }

    public static Transition changeBoundsDecelerate(long duration) {
        changeBounds = new ChangeBounds();
        changeBounds.setInterpolator(new DecelerateInterpolator());
        changeBounds.setDuration(duration);
        return changeBounds;
    }

    public static Transition fade(long duration) {
        Fade fade = new Fade();
        fade.setDuration(duration);
        return fade;
    }

    public static Transition explode(long duration) {
        Explode explode = new Explode();
        explode.setDuration(duration);
        return explode;
    }
}

File: coreutils/src/main/java/com/tanveershafeeprottoy/coreutils/TransitionUtils.kt
package com.tanveershafeeprottoy.coreutils

import android.transition.ChangeBounds
import android.transition.Explode
import android.transition.Fade
import android.transition.Transition
import android.view.animation.DecelerateInterpolator

object TransitionUtils {
    private var changeBounds: ChangeBounds? = null

    fun changeBounds(duration: Long): Transition {
        changeBounds = ChangeBounds()
        changeBounds?.duration = duration
        return changeBounds!!
    }

    fun changeBoundsDecelerate(duration: Long): Transition {
        changeBounds = ChangeBounds()
        changeBounds?.interpolator = DecelerateInterpolator()
        changeBounds?.duration = duration
        return changeBounds!!
    }

    fun fade(duration: Long): Transition {
        val fade = Fade()
        fade.duration = duration
        return fade
    }

    fun explode(duration: Long): Transition {
        val explode = Explode()
        explode.duration = duration
        return explode
    }
}